robots/purePursuitController.h

```diff
@@ -0,0 +1,169 @@
+/*
+    Pure Pursuit Controller. The controller takes a list of waypoints as a path to follow,
+    then it calculates the optimal turning angle to stay on the path
+*/
+
+// BoB robotics includes
+#include "common/main.h"
+#include "robots/car_display.h"
+
+// Third-party includes
+#include "third_party/units.h"
+
+// Standard C++ includes
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+// sign function
+template <typename T> int sgn(T val) {
+    return (T(0) < val) - (val < T(0));
+}
+
+namespace BoBRobotics {
+namespace Robots {
+
+using namespace units::math;
+using namespace units::literals;
+
+class PurePursuitController {
+
+    using millimeter_t = units::length::millimeter_t;
+    using degree_t = units::angle::degree_t;
+    using radian_t = units::angle::radian_t;
+   
+public:
+
+    PurePursuitController() {
+
+    }
+
+    PurePursuitController(millimeter_t lookahead, millimeter_t wheelBaseLength) : 
+                          lookAheadDistance(lookahead), wheelBase(wheelBaseLength) {
+
+    }
+
+    //! set waypoints which forms a path to be followed
+    void setWayPoints(std::vector<Vector2<millimeter_t>> wp) {
```

Comment:
Again, these should be passed as ``const std::vector<Vector2<millimeter_t>> &wp``